### PR TITLE
Enumerable methods

### PR DESCRIPTION
**Pull request by andresporras3423**
* Extending Enumerable module to add more methods that received both arrays and ranges
* Extension of Enumerable module in enumerable-methods file
* Multiply_els in main.rb
* Using yield and call method as it was required
* my_inject method receive the next list of symbols (+, -, *, /, **, & and ||)

### CODE REVIEW

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  end
+
+  def my_all?
+    return (my_all? { |x| !x.nil? }) unless block_given?
+
+    my_each do |value|
+      return false unless yield value
+    end
+    true
+  end
```
Comment (shshamim63) [inline enumerable_methods.rb]: #### Mandatory
my_all?` identical and returns the same thing as ruby's [all?](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
1. When no block or argument is given returns true when none of the collection members are false or nil. For an example ```[1, 2, false].my_all?``` should return false.
2. When a class is passed as an argument returns true if all of the collection is a member of such class.
```ruby
[1, 2, 3].my_all?(Integer) #should return true
```
3. When a Regex is passed as an argument returns true if all of the collection matches the Regex.
```ruby
%w[dog door rod blade].my_all?(/d/)  #should return true
```
4. When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
```ruby
[1, 1, 1].my_all?(1) #should return true
```

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  def my_any?
+    return (my_any? { |x| !x.nil? }) unless block_given?
+
+    my_each do |value|
+      return true if yield value
+    end
+    false
+  end
```
Comment (shshamim63) [inline enumerable_methods.rb]: #### Mandatory
my_any?` identical and returns the same thing as ruby's [any?](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). 
`([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
1. When no block or argument is givenreturns true if at least one of the collection is not false or nil. For an example ```[1, false, false].my_any?``` should return true.
2. When a class is passed as an argument returns true if at least one of the collection is a member of such class.
```ruby
[1, false, 'demo'].my_any?(String) #should return true
```
3. When a Regex is passed as an argument returns true only if none of the collection matches the Regex.
```ruby
%w[dog door rod blade].my_any?(/d/)  #should return true
```
4. When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```ruby
[1, 3, 5].my_none?(1) #should return true
```

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  def my_none?
+    return (my_none? { |x| !x.nil? }) unless block_given?
+
+    my_each do |value|
+      return false if yield value
+    end
+    true
+  end
```
Comment (shshamim63) [inline enumerable_methods.rb]: #### Mandatory
my_none?` identical and returns the same thing as ruby's [none?](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). 
`([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
1. When no block or argument is given returns true only if none of the collection members is true. For an example ```[nil, false, false].my_none?``` should return true.
2. When a class is passed as an argument returns true if none of the collection is a member of such class.
```ruby
[1, 2, 3].my_all?(String) #should return true
```
3. When a Regex is passed as an argument returns true only if none of the collection matches the Regex.
```ruby
%w[dog door rod blade].my_none?(/z/) #should return true
```
4. When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
```ruby
[2, 3, 5].my_none?(1) #should return true
```

File: enumerable_methods.rb [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      my_each do |value|
+        counter += 1 if nval == value
+      end
+    else
+      my_each do
+        counter += 1
+      end
+    end
+    counter
+  end
```
Comment (shshamim63) [inline enumerable_methods.rb]: ##### Mandatory
`my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
1. Counts the number of elements yielding a true value if a block is given
```ruby
[1, 2, false].my_count { |element| element.is_a?(Integer) }  #should return 2
```

**Review by shshamim63 — COMMENTED** [review]
## Status :spades: Changes Required :spades: 
___
Hi @andresporras3423 :wave: 
Awesome initiative. code looks neat and clean. You have declared a main.rb file which contains user friendly component to test multiply_els method. :clap: But still there are some issues that are blocking me to approve this project. Please follow the comments below to complete this project.
___
1. my_all?
2. my_any?
3. my_none?
4. my_count
___
##### After resolving all the issues please send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by andresporras3423** [comment]
Hi shshamim63,

Thanks for your feedback

I have made the next changes:
* modify my_any? to receive regex and classes. return false if no item is different from nil or false
* modify my_all? to receive regex and classes. return false if not all the items are different from nil or false
* modify my_none? to receive regex and classes. return true if not all the items are different from nil or false
* modify my_count to receive a block

**Review by fatymahmed — APPROVED** [review]
## Hi @andresporras3423,
### Great job making the suggested changes🎉🎉
### It's a 👍🏾 from me!
## ✔️ Approved!

[OPTIONAL]
- Please don't change stickler config files, see [a screenshot of requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from Pathwright. The only exception is:
> you can have Style/CaseEquality cop disabled as described [here](https://rubydoc.info/gems/rubocop/0.24.1/frames#Disabling_Cops_within_Source_Code), see [Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/style_case_equality.png) of requirement from Pathwright.

Since this wasn't mentioned in the first review, it's marked as **optional**, but it's highly recommended that you make this change before merging.

Good luck with your next projects.
 _Don't forget to submit your project completion form_

**Comment by andresporras3423** [comment]
Thank you Fatima

**Pull request merged**